llvm/include/llvm/CodeGen/TargetSubtargetInfo.h

```diff
@@ -338,6 +338,11 @@ class TargetSubtargetInfo : public MCSubtargetInfo {
   /// the pass, with architecture specific overrides providing the information
   /// where they are implemented.
   virtual bool supportsInitUndef() const { return false; }
+
+  /// For \p F, call \p Body with the name and value of each launch bound.
+  virtual void forEachLaunchBound(
+      const Function &F,
+      std::function<void(StringRef Name, unsigned Value)> Body) const {}
```

Comment:
It would be more appropriate to add something to TargetTransformInfo. It also needs a more specific name and documentation comment. Also should use int64_t for arbitrary integer values 

This can also be function_ref

Reply:
> It would be more appropriate to add something to TargetTransformInfo.

Done.

> It also needs a more specific name and documentation comment.

I'm fine to change it, but what details do you want the name and comment to reveal?

> Also should use int64_t for arbitrary integer values
> 
> This can also be function_ref

Done.

Reply:
Everything appears to be addressed except [the part that continues here](https://github.com/llvm/llvm-project/pull/102944#discussion_r1773738493).  Resolving.